lib/core/scheduler.rb

```diff
@@ -119,6 +112,13 @@ def sync_triggered?(connector_settings)
         return false
       end
 
+      # We want to sync when sync never actually happened
+      last_synced = connector_settings[:last_synced]
+      if last_synced.nil? || last_synced.empty?
```

Comment:
```suggestion
      if scheduling_settings[:interval].present? && (last_synced.nil? || last_synced.empty?)
```

I think this is already logically implied, but I think it's easier to reason about if we make this explicit.

Reply:
With the latest commit https://github.com/elastic/connectors-ruby/commit/59f526ff017e38da596851800bc7f4508f328552, this is not needed.